lib/Sources/BluetoothAction/Options+Decode.swift

```diff
@@ -127,13 +127,15 @@ extension [JsType] {
     func compactMapToUUIDs() -> [UUID]? {
         self.compactMap { $0.string.toUuid() }
     }
+}
 
+extension Dictionary where Key == String, Value == JsType {
     func compactMapToUint8Array() -> [UInt8]? {
-        self.compactMap { $0.number }.compactMap { UInt8(truncating: $0) }
+        self.sorted(by: { $0.key < $1.key }).compactMap { $0.value.number }.compactMap { UInt8(truncating: $0) }
     }
 
     func compactMapToUint16Array() -> [UInt16]? {
-        self.compactMap { $0.number }.compactMap { UInt16(truncating: $0) }
+        self.sorted(by: { $0.key < $1.key }).compactMap { $0.value.number }.compactMap { UInt16(truncating: $0) }
```

Comment:
Same comment about the lambda and combining the operation as above applies here.